clang/include/clang/AST/Expr.h

```diff
@@ -2037,6 +2037,112 @@ class PredefinedExpr final
   }
 };
 
+// This represents a use of the __builtin_unique_stable_name, which takes either
+// a type-id or an expression, and at CodeGen time emits a unique string
+// representation of the type (or type of the expression) in a way that permits
+// us to properly encode information about the SYCL Kernels.
```

Comment:
```suggestion
// us to properly encode information about the SYCL kernels.
```
May want to drop mention of SYCL only because this functionality seems useful outside of SYCL as well.